udata_gouvfr/tasks.py

```diff
@@ -0,0 +1,49 @@
+import mongoengine
+import requests
+
+from collections import defaultdict
+from flask import current_app
+
+from udata.commands import success, error
+from udata.core.dataset.models import Dataset
+from udata.tasks import job
+
+from udata_gouvfr import APIGOUVFR_EXTRAS_KEY
+
+
+def get_dataset(id_or_slug):
+    obj = Dataset.objects(slug=id_or_slug).first()
+    return obj or Dataset.objects.get(id=id_or_slug)
+
+
+def process_dataset(d_id, apis):
+    try:
+        dataset = get_dataset(d_id)
+    except (Dataset.DoesNotExist, mongoengine.errors.ValidationError):
+        return error(f'Dataset {d_id} not found')
+    dataset.extras[APIGOUVFR_EXTRAS_KEY] = apis
+    dataset.save()
+    success(f'Imported {len(apis)} API(s) for {str(dataset)}')
+
+
+@job('apigouvfr-load-apis')
+def apigouvfr_load_apis(self):
+    '''Load dataset-related APIs from api.gouv.fr'''
+    r = requests.get(current_app.config['APIGOUVFR_URL'])
+    r.raise_for_status()
+
+    apis = r.json()
+    datasets_apis = defaultdict(list)
+    for api in apis:
+        # strictly non open APIs, ignore
+        if api['is_open'] == -1:
```

Comment:
Not your fault but I would have expected a boolean. Is it going to change? Maybe anticipate the change with

```suggestion
        if api['is_open'] in [-1, False]:
```

Reply:
Nope, three states needed (open, semi-open (token required), closed).

Reply:
Hi @Antoine, 

I admit it is far from perfect. 

`is_open` used to be a Boolean. I recently switched it to [-1, 0, 1]. Because we have a few APIs (INSEE, INPI) that are not really open. 

They require you to create an account, and sometimes justify why you want to access the API. According to Etalab, this is not an open API.

So I created three states : 
-1 : this API require to request access in order to use it
0 : this API require an account
1 : this API is fully open and you can access its data

This is misleading and error prone. :/ My bad. I should maybe hide this part which is some internal logic and expose a Boolean. @abulte what do you think ?

Reply:
I don't mind. Maybe a string would be self-documenting and thus less misleading, but I like having the three states exposed. We could use this info on the card at some point.

Reply:
Was about to write the same thing than @abulte. 3 states are fine in a different key name, `is_open` without a boolean value is misleading

Reply:
Ok then as @abulte suggested I switch to : 
`is_open` : `open` | `closed`| `semi_open`

Reply:
If we can down this road, maybe `openness` (or smtg) with strings, and `is_open` with a boolean (or not at all).

Reply:
https://api.gouv.fr/api/v1/apis
I will remove the is_open field once you have updated your code :)

Reply:
Go ahead, it won't break anything ;-)